recipes/default.rb

```diff
@@ -34,26 +34,91 @@
   not_if { node["kagent"]["enabled"] == "false" }
 end
 
-# Generate an API key 
-api_key = ""
+# Generate an API key
+ruby_block 'fetch-api-key' do
+  block do
+    require 'net/https'
+    require 'http-cookie'
+    require 'json'
 
-# Template the configuration file 
+    hopsworks_fqdn = consul_helper.get_service_fqdn("hopsworks.glassfish")
+    hopsworks_endpoint = "https://#{hopsworks_fqdn}:#{node['hopsworks']['internal']['port']}"
+    url = URI.parse("#{hopsworks_endpoint}/hopsworks-api/api/auth/service")
+    api_key_url = URI.parse("#{hopsworks_endpoint}/hopsworks-api/api/users/apiKey")
+
+    params =  {
+      :email => node['onlinefs']['hopsworks']['email'],
+      :password => node['onlinefs']['hopsworks']["password"]
+    }
+
+    api_key_params = {
+      :name => "onlinefs",
+      # check with fabio which scopes are needed
+      :scope => "FEATURESTORE,KAFKA,PROJECT"
```

Comment:
I think just Kafka and Project, but the name needs to be random, otherwise on multiple instances  will try to generate the same API key.

Reply:
hm users will see it, so I will prefix it at least with `online_fs`?